modules/core/src/smithy4s/Transformation.scala

```diff
@@ -119,4 +119,30 @@ object Transformation {
       }
     }
 
+  implicit def mappedHintsServiceTransform[Alg[_[_, _, _, _, _]]]: Transformation[Hints=>Hints, Service[Alg], Service[Alg]] =
+    (func: Hints => Hints, that: Service[Alg]) => new Service[Alg] {
+      override type Operation[I, E, O, SI, SO] = that.Operation[I, E, O, SI, SO]
+
+      override def endpoints: List[Endpoint[_, _, _, _, _]] = that.endpoints.map(_.mapHints(func))
+
+      override def endpoint[I, E, O, SI, SO](op: Operation[I, E, O, SI, SO]): (I, Endpoint[I, E, O, SI, SO]) = {
+        that.endpoint(op) match {
+          case (i, endpoint) => (i, endpoint.mapHints(func))
+        }
+      }
```

Comment:
I feel like this might benefit from some caching of the `endpoint.mapHints` call

Reply:
good point , i think ```endpoints``` would also benefit from that

Reply:
You could make `endpoints` a val :)

Reply:
Good point , forgot about that thing